MODULE.md

```diff
@@ -0,0 +1,1200 @@
+                      │ and integrate modules   │
+                      └───────────┬─────────────┘
+                                  │
+                                  │ Provisions
+                                  ▼
+                      ┌─────────────────────────┐
+                      │      Workspaces         │◄─── Developers use
+                      │                         │     workspaces
+                      │  Running development    │
+                      │     environments        │
+                      └─────────────────────────┘
+```
+
+### Terraform Foundation
+
+Coder modules are implemented as Terraform modules using HashiCorp's infrastructure-as-code framework. The [Coder Terraform provider](https://registry.terraform.io/providers/coder/coder/latest/docs) extends standard Terraform functionality with workspace-specific resources:
+
+**Core Module Resources:**
+
+- [`coder_script`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/script): Executes scripts during workspace lifecycle events
+- [`coder_app`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/app): Creates accessible applications within the workspace interface
+- [`coder_env`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/env): Sets environment variables in the workspace
+
+**Template Resources (rarely used in modules):**
+
+- [`coder_agent`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/agent): Defines agent configuration for workspace runtime
+- [`coder_parameter`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/parameter): Collects user input during workspace provisioning
+- [`coder_metadata`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/metadata): Displays resource information in the workspace interface
+
+**Advanced Resources:**
+
+- [`coder_agent_instance`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/agent_instance): Manages agent instance connections
+- [`coder_ai_task`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/ai_task): Defines AI-powered task automation
+- [`coder_devcontainer`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/devcontainer): Configures development container environments
+
+Modules use standard Terraform syntax plus these Coder-specific resources to integrate with workspaces.
+
+### Provider and Terraform Requirements
+
+Modules specify required Terraform and provider versions:
+
+```tf
+terraform {
+  required_version = ">= 1.0"
+
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = ">= 2.5"
+    }
+  }
+}
+```
+
+This ensures compatibility with minimum versions of Terraform and the Coder provider.
+
+**Current Version Standards:**
+
+- **Latest Coder Provider**: v2.8.0 (June 2025)
+- **Recommended Minimum**: `>= 2.5` (used by most current modules)
```

Comment:
```suggestion
- **Recommended Minimum**: `>= 2.5` (used by most current modules)
```
There is no recommended version. It always depends on the feature you are using in your module and the minimum version of the provider where that feature is available.